src/compiler/transformers/esDecorators.ts

```diff
@@ -2377,10 +2359,18 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
         );
     }
 
-    function createSymbolMetadataReference() {
-        return factory.createElementAccessExpression(
-            factory.createSuper(),
+    function createSymbolMetadataReference(classSuper: Identifier) {
+        const classSuperSymbolMetadata = factory.createElementAccessExpression(
+            classSuper,
             factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), "metadata"),
         );
+        return factory.createConditionalExpression(
+            factory.createLogicalOr(
+                factory.createTypeCheck(classSuperSymbolMetadata, "object"),
+                factory.createTypeCheck(classSuperSymbolMetadata, "function"),
+            ),
+            factory.createToken(SyntaxKind.QuestionToken), classSuperSymbolMetadata,
+            factory.createToken(SyntaxKind.ColonToken), factory.createNull(),
+        );

```

Comment:
I'm not sure we need the `typeof` test here, the spec doesn't do that and evaluating `_classSuper[Symbol.metadata]` twice has the potential to trigger getters twice.

Just using `Object.create(_classSuper[Symbol.metadata] ?? null)` should be more than adequate.

Reply:
See the suggestion here: https://github.com/microsoft/TypeScript/pull/54657/files/da43f1f7c6178f42cc2e973ffa4e32a1d963c116..2996d8ea2c9d08c086ae4367ef3469630cb7cae8#r1238995785